src/epgsql_sock.erl

```diff
@@ -120,8 +120,12 @@ handle_cast({{Method, From, Ref}, Command} = Req, State)
 handle_cast(stop, State) ->
     {stop, normal, flush_queue(State, {error, closed})};
 
-handle_cast(cancel, State = #state{backend = {Pid, Key}}) ->
-    {ok, {Addr, Port}} = inet:peername(State#state.sock),
+handle_cast(cancel, State = #state{backend = {Pid, Key},
+                                   sock = TimedOutSock}) ->
+    {ok, {Addr, Port}} = case State#state.mod of
+                             gen_tcp -> inet:peername(TimedOutSock);
+                             ssl -> ssl:peername(TimedOutSock)
+                         end,
     SockOpts = [{active, false}, {packet, raw}, binary],
     %% TODO timeout
     {ok, Sock} = gen_tcp:connect(Addr, Port, SockOpts),
```

Comment:
In my tests, Postgres accepted cancel commands over TCP even if they were canceling a query that had been issued over SSL. [The manual](http://www.postgresql.org/docs/9.3/static/protocol-flow.html#AEN99368) doesn't seem to cover what to do here. As I read it, [libpq doesn't use SSL either](https://github.com/postgres/postgres/blob/080eabe2e8a184ff40b7380aaaa9418714acace9/src/interfaces/libpq/fe-connect.c#L3240). Thoughts?